zerver/tests/test_messages.py

```diff
@@ -1300,40 +1295,37 @@ def test_topic_links_use_stream_realm(self) -> None:
         realm_filter = RealmFilter(realm=zulip_realm,
                                    pattern=r"#(?P<id>[0-9]{2,8})",
                                    url_format_string=url_format_string)
-        realm_filter.save()
         self.assertEqual(
             realm_filter.__str__(),
             '<RealmFilter(zulip): #(?P<id>[0-9]{2,8})'
             ' https://trac.zulip.net/ticket/%(id)s>')
 
-        message_from_zulip = Message(
-            sender=othello,
-            recipient=recipient,
-            content='hello world',
-            date_sent=timezone_now(),
-            sending_client=sending_client,
-            last_edit_time=timezone_now(),
-            edit_history='[]'
-        )
-        message_from_zulip.set_topic_name(topic_name)
-        message_from_zulip.save()
-        message_from_lear = Message(
-            sender=cordelia,
-            recipient=recipient,
-            content='hello world',
-            date_sent=timezone_now(),
-            sending_client=sending_client,
-            last_edit_time=timezone_now(),
-            edit_history='[]'
-        )
-        message_from_lear.set_topic_name(topic_name)
-        message_from_lear.save()
-
-        dct = MessageDict.to_dict_uncached_helper(message_from_zulip)
-        self.assertEqual(dct[TOPIC_LINKS], [url])
-        dct = MessageDict.to_dict_uncached_helper(message_from_lear)
-        self.assertEqual(dct[TOPIC_LINKS], [url])
-        self.assertNotEqual(lear_realm, zulip_realm)
+        def get_message(sender: UserProfile) -> Message:
+            msg = Message(
+                sender=sender,
+                recipient=recipient,
+                content='hello world',
+                date_sent=timezone_now(),
+                sending_client=sending_client,
+                last_edit_time=timezone_now(),
+                edit_history='[]'
+            )
+            msg.set_topic_name(topic_name)
+            msg.save()
+            return msg
```

Comment:
I'd prefer to generate messages using e.g. `self.send_stream_message`; we try to avoid just creating database objects in non-standard ways in the test code, since it can result in the test not testing the actual message situations we care about.  I think that should end up being less code as well.